### PR TITLE
Hw 21

### PR DESCRIPTION
**Pull request by Sumned**

### CODE REVIEW

**Review by dmytro-hryhoruk — COMMENTED** [review]
https://www.youtube.com/watch?v=BHSsWTbuow4 configure travis ci

File: src/main/java/mate/academy/internetshop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                    connection.prepareStatement(query, PreparedStatement.RETURN_GENERATED_KEYS);
+            statement.setString(1, element.getName());
+            statement.setBigDecimal(2, element.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            resultSet.next();
+            element.setId(resultSet.getLong(1));
+            return element;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to create " + element.getName());
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/dao/jdbc/ProductDaoJdbcImpl.java]: you are catching SQLException in your controllers but throwing DataProcessingException here

File: src/main/java/mate/academy/internetshop/service/impl/ShoppingCartServiceImpl.java [hunk 2]
```diff
@@ -44,8 +45,14 @@ public ShoppingCart getByUserId(Long userId) {
...
                 .findFirst()
-                .orElseGet(() -> shoppingCartDao.create(
-                        new ShoppingCart(new ArrayList<>(), userService.get(userId))));
+                .orElseGet(() -> {
+                    try {
+                        return shoppingCartDao.create(
+                                new ShoppingCart(new ArrayList<>(), userService.get(userId)));
+                    } catch (SQLException e) {
+                        throw new RuntimeException(e);
+                    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/service/impl/ShoppingCartServiceImpl.java]: your shoppingCart should be created by that time along with user

File: src/main/resources/init_db.sql [hunk 3]
```diff
@@ -0,0 +1,7 @@
+CREATE SCHEMA `internet_shop` ;
+CREATE TABLE `internet_shop`.`products` (
+  `product_id` BIGINT(11) NOT NULL AUTO_INCREMENT,
+  `product_name` VARCHAR(225) NOT NULL,
+  `product_price` DECIMAL(11) NOT NULL,
+  PRIMARY KEY (`product_id`),
+  UNIQUE INDEX `product_name_UNIQUE` (`product_name` ASC) VISIBLE);
```
Comment (dmytro-hryhoruk) [inline src/main/resources/init_db.sql]: empty line

File: src/main/java/mate/academy/internetshop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/mate/academy/internetshop/dao/jdbc/ProductDaoJdbcImpl.java]: let's catch your custom exception in your controllers

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+language: java
+jdk:
+  - openjdk11
+script:
+  - mvn clean install
+cache:
+  directories:
+    - $HOME/.m2
```
Comment (dmytro-hryhoruk) [inline .travis.yml]: empty line

File: src/main/java/mate/academy/internetshop/controller/product/GetAllProductController.java [hunk 5]
```diff
@@ -8,10 +8,14 @@
 import javax.servlet.http.HttpServletResponse;
 import mate.academy.internetshop.lib.Injector;
 import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.model.Role;
 import mate.academy.internetshop.service.ProductService;
+import mate.academy.internetshop.service.UserService;
 
 public class GetAllProductController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/controller/product/GetAllProductController.java]: you should have two different controllers: one for users and another for admins

File: src/main/java/mate/academy/internetshop/controller/product/GetAllProductController.java [hunk 6]
```diff
@@ -20,6 +24,14 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
         List<Product> allProducts = productService.getAll();
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/controller/product/GetAllProductController.java]: doesn't getAll() method throws DataProcessingException? or you are handling it somehow in service?

File: src/main/java/mate/academy/internetshop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/mate/academy/internetshop/dao/jdbc/ProductDaoJdbcImpl.java]: let's add exception object to new Exception

**Review by dmytro-hryhoruk — APPROVED** [review]

File: .travis.yml (hunk 4 above)
Comment (Sumned) [reply .travis.yml]: +

**Pull request merged**